### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by saubyk**
## Change Description
CODEOWNERS file has been added under .github folder to enable automated review assignments based on ownership of different areas of the code base.

Doc Reference: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ellemouton) [inline .github/CODEOWNERS:72]: can I be added as a secondary here & to `routing` too? im getting more experience in those now with all the gossip work

File: .github/CODEOWNERS
Comment (ellemouton) [inline .github/CODEOWNERS:141]: happy to be here too

File: .github/CODEOWNERS
Comment (ellemouton) [inline .github/CODEOWNERS:15]: happy to be here but just a heads up that i have no experience here. Feels like perhaps @bitromortac would be a better fit here?

File: .github/CODEOWNERS
Comment (ellemouton) [inline .github/CODEOWNERS:45]: happy to be secondary here too

**Review by ellemouton — COMMENTED** [review]
Nice - will be interesting to see how this "round-2" of the code owners file goes. Spoke about this with @Roasbeef some time ago and I think iirc the reason the first round didnt work is that since some files are touched way more than others - some people's review queues just got overwhelmingly large. But I think that miiight be because back then there were also defaults (ie, if no-one ones this pkg, assign x & y to review).

File: .github/CODEOWNERS
Comment (guggero) [inline .github/CODEOWNERS:6]: can add me here

**Review by guggero — APPROVED** [review]
Nice, let's try this again :+1:

File: .github/CODEOWNERS
Comment (saubyk) [reply .github/CODEOWNERS:1]: added

File: .github/CODEOWNERS
Comment (saubyk) [reply .github/CODEOWNERS:201]: added

File: .github/CODEOWNERS
Comment (Roasbeef) [reply .github/CODEOWNERS:1]: So I don't think it needs to be 100% exhaustive. Otherwise, those ppl will be added each time a contributor makes a typo fix in a proto. Certain packages like `build`, have no true owner, and instead we can rely on the load balancing option for code review. 

If we want to chop up the sub-server packages/files, then I think that could make sense.

Comment (saubyk) [reply .github/CODEOWNERS:1]: So far, I haven't included docs, input, labels, lnmock, make, scripts, subscribe and tools.
If any of these should be included, or any that you think is best excluded so that reviews are assigned via load balancing option for those, do let me know.

**Review by yyforyongyu — APPROVED** [review]

**Review by yyforyongyu — COMMENTED** [review]
Could you squash the commits? thanks 🐻‍❄️

**Review by ProofOfKeags — APPROVED** [review]

**Review by ellemouton — COMMENTED** [review]
lgtm once commits are squashed :)

**Pull request merged**
